pkg/cli/clients/management.go

```diff
@@ -1022,9 +1022,46 @@ func isResourceInApplication(resource generated.GenericResource, applicationID s
 }
 
 func isResourceInEnvironment(resource generated.GenericResource, environmentID string) bool {
-	obj, found := resource.Properties["environment"]
-	// A resource may not have an environment associated with it.
+	obj, found := resource.Properties["environment"] // A resource may not have an environment associated with it.
 	if !found {
+		// Special case for environment resources themselves
+		if resource.Type != nil && strings.EqualFold(*resource.Type, "Applications.Core/environments") && resource.ID != nil {
+			// First try direct comparison
+			if strings.EqualFold(*resource.ID, environmentID) {
+				return true
+			}
+
+			// If no direct match, check if one is a shortname and the other is a full ID
+			shortNameEnv := environmentID
+			if strings.Contains(environmentID, "/") {
+				parts := strings.Split(environmentID, "/")
+				shortNameEnv = parts[len(parts)-1]
+			}
+
+			shortNameResource := *resource.ID
+			if strings.Contains(*resource.ID, "/") {
+				parts := strings.Split(*resource.ID, "/")
+				shortNameResource = parts[len(parts)-1]
+			}
+
+			if shortNameEnv != "" && shortNameResource != "" &&
+				strings.EqualFold(shortNameEnv, shortNameResource) {
+				return true
+			}
+		}
+
+		// Special handling for the default environment - resources without an explicit
+		// environment property are considered to be in the default environment
```

Comment:
Is it possible for a resource to not have an environment?

Reply:
Based off the code in main, I believe so: https://github.com/radius-project/radius/blob/main/pkg/cli/clients/management.go#L1026